django/db/utils.py

```diff
@@ -276,3 +276,24 @@ def get_migratable_models(self, app_config, db, include_auto_created=False):
         """Return app models allowed to be migrated on provided db."""
         models = app_config.get_models(include_auto_created=include_auto_created)
         return [model for model in models if self.allow_migrate_model(db, model)]
+
+
+class Text(str):
```

Comment:
I'm not sure a postgresql-specific class should live here... perhaps if it was called `PgText` or similar?

Reply:
Yeah we can easily shuffle that around. But since the usages are not in the backend itself we cannot easily move it into the backend because that would require psycopg to be available. We could import directly in `as_posgtresql` but that is not nice either imo.

Reply:
This entirely removed by https://github.com/apollo13/django/pull/3 which the foundational parts have been extracted in https://github.com/django/django/pull/16245. I suspect this is one or the origin of the reported performance regression.